Xamarin.Forms.Core/Shell/ShellNavigatingEventArgs.cs

```diff
@@ -6,12 +6,19 @@ namespace Xamarin.Forms
 {
 	public class ShellNavigatingEventArgs : EventArgs
 	{
-		int _deferalCount;
+		int _deferralCount;
 		Func<Task> _deferralFinishedTask;
 		TaskCompletionSource<bool> _deferredTaskCompletionSource;
+		bool _deferralCompleted = false;
 
 		public ShellNavigatingEventArgs(ShellNavigationState current, ShellNavigationState target, ShellNavigationSource source, bool canCancel)
 		{
+
+#if NETSTANDARD2_0
+			_deferralFinishedTask = () => Task.CompletedTask;
```

Comment:
This is the main fix. Just initializing the deferral finished task to a completed task so if no delayed task gets registered the code doesn't crash with an NRE